cannon/mipsevm/tests/evm_common_test.go

```diff
@@ -499,24 +499,31 @@ func TestEVMFault(t *testing.T) {
 		name   string
 		nextPC uint32
 		insn   uint32
+		errMsg string
 	}{
-		{"illegal instruction", 0, 0xFF_FF_FF_FF},
-		{"branch in delay-slot", 8, 0x11_02_00_03},
-		{"jump in delay-slot", 8, 0x0c_00_00_0c},
+		{"illegal instruction", 0, 0xFF_FF_FF_FF, "invalid instruction"},
+		{"branch in delay-slot", 8, 0x11_02_00_03, "branch in delay slot"},
+		{"jump in delay-slot", 8, 0x0c_00_00_0c, "jump in delay slot"},
 	}
 
 	for _, v := range versions {
 		for _, tt := range cases {
 			testName := fmt.Sprintf("%v (%v)", tt.name, v.Name)
 			t.Run(testName, func(t *testing.T) {
-				goVm := v.VMFactory(nil, os.Stdout, os.Stderr, testutil.CreateLogger(), testutil.WithNextPC(tt.nextPC))
+				// set the return address ($ra) to jump into when test completes
+				goVm := v.VMFactory(nil, os.Stdout, os.Stderr, testutil.CreateLogger(), testutil.WithNextPC(tt.nextPC), testutil.WithRegisters(31, testutil.EndAddr))
+				proofData := v.ThreadProofEncoder(nil, os.Stdout, os.Stderr, testutil.CreateLogger(), testutil.WithNextPC(tt.nextPC), testutil.WithRegisters(31, testutil.EndAddr))
 				state := goVm.GetState()
 				state.GetMemory().SetMemory(0, tt.insn)
-				// set the return address ($ra) to jump into when test completes
-				state.GetRegistersRef()[31] = testutil.EndAddr
-
+				statePCs := []uint32{state.GetPC()}
+				// TODO: Hardcode here to achieve the expected result.
+				// Future improvements might involve having the Go VM indicate attempted memory accesses
+				// and caching this info to generate proofs for the solidity VM.
+				if tt.errMsg == "invalid instruction" {
+					statePCs = append(statePCs, 0xa7ef00cc)
```

Comment:
This looks like `EndAddr - 4`?  Where does this value come in?

Reply:
Aha this is a tricky case, I got this from @Inphi   

>"The EVM reverts from here - https://github.com/ethereum-optimism/optimism/blob/1217d4ac74e4dfddc23391f41e50e51f7c64a772/packages/contracts-bedrock/src/cannon/libraries/MIPSInstructions.sol#L125
Basically, the instruction decoder is sorta  buggy. It admits invalid instructions and attempts to perform a memory read for the 0xff opcode.
So if you were to provide an additional  valid memory proof, it should eventually reafch the codepath that detects an invalid instruction.
Basically, all opcodes >= 0x20 are readed as memory access opcodes. And we provide a 0xff for that test
The way to go here would be to append a stubbed, but valid, memory proof for some random address"

Reply:
ah - thanks for the explanation!